### PR TITLE
Return value of compare 1 if differences

### PR DESCRIPTION
**Pull request by clementval**
For integration test, otherwise jenkins thinks everythings is fine :-)

### CODE REVIEW

**Comment by clementval** [comment]
I'll do the change in Serialbox2 as well once this is ok.

File: tools/compare.cpp [hunk 1]
```diff
@@ -241,10 +243,15 @@ int compare(const std::string& directory1, const std::string& basename1,
 				std::cerr << "Unsupported type: " << info1.type() << std::endl;
 				return 2;
 			}
+                        hasDifferences = hasDifferences | !equal;
 		}
 	}
 
 	// return (int) (!equal);
```
Comment (havogt) [inline tools/compare.cpp]: Can be deleted?

File: tools/compare.cpp [hunk 2]
```diff
@@ -175,6 +175,8 @@ int compare(const std::string& directory1, const std::string& basename1,
 
 	vector<Savepoint> savepoints = serializer1.savepoints();
 
+        bool hasDifferences = false;
```
Comment (havogt) [inline tools/compare.cpp]: indendation looks wrong for all the lines which you added.

**Review by havogt — CHANGES_REQUESTED** [review]

**Review by clementval — COMMENTED** [review]
@havogt This file is using sadly using tabs. If you don't mind, I would like to push a commit with changes to spaces instead...

**Comment by havogt** [comment]
Oh yes of course. I don't want to be responsible that someone's salary has to be cut because of the tabs!

**Review by havogt — APPROVED** [review]
I didn't check all the reformatting...

**Comment by clementval** [comment]
Thanks! By the way, do we have a code style for serialbox yet?

**Comment by havogt** [comment]
(For serialbox2 (C++ part) there is a clang-format style file.)

**Comment by clementval** [comment]
Ok so I suggest that for the old one we don't care too much as we will switch anyway ...

**Pull request merged**